OricExplorer/Classes/OricTape.cs

```diff
@@ -226,67 +212,76 @@ private ProgramHeader ReadProgramHeader()
+                    highByte = m_bTapeBuffer[m_ui32BufferIdx];
+                    lowByte = m_bTapeBuffer[m_ui32BufferIdx + 1];
 
-            programHeader.ui16StartAddress = Convert.ToUInt16(lowByte + (highByte * 256));
+                    programHeader.ui16StartAddress = Convert.ToUInt16(lowByte + (highByte * 256));
 
-            if(bProgramFormat == 0x00)
-                programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.BasicProgram;
-            else
-            {
-                if(programHeader.ui16StartAddress == 0xA000)
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.HiresScreen;
-                else if(programHeader.ui16StartAddress == 0xB500)
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.CharacterSet;
-                else if(programHeader.ui16StartAddress.In((ushort)0xBB80, (ushort)0xBBA8))
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.TextScreen;
-                else
-                    programHeader.bProgramFormat = (byte)OricProgram.ProgramFormat.BinaryFile;
-            }
+                    if (bProgramFormat == 0x00)
+                        programHeader.Format = OricProgram.ProgramFormat.BasicProgram;
+                    else
+                    {
+                        if (programHeader.ui16StartAddress == 0xA000)
+                            programHeader.Format = OricProgram.ProgramFormat.HiresScreen;
+                        else if (programHeader.ui16StartAddress == 0xB500)
+                            programHeader.Format = OricProgram.ProgramFormat.CharacterSet;
+                        else if (programHeader.ui16StartAddress.In((ushort)0xBB80, (ushort)0xBBA8))
+                            programHeader.Format = OricProgram.ProgramFormat.TextScreen;
+                        else
+                            programHeader.Format = OricProgram.ProgramFormat.BinaryFile;
+                    }
 
-            m_ui32BufferIdx += 3;
+                    m_ui32BufferIdx += 3;
 
-            while(m_bTapeBuffer[m_ui32BufferIdx] != 0)
-            {
-                programHeader.strProgramName += Convert.ToChar(m_bTapeBuffer[m_ui32BufferIdx]);
-                m_ui32BufferIdx++;
-            }
+                    uint uintMaxBufferIdx = m_ui32BufferIdx + 16;
+                    while (m_bTapeBuffer[m_ui32BufferIdx] != 0 && m_ui32BufferIdx < uintMaxBufferIdx)
+                    {
+                        programHeader.strProgramName += Convert.ToChar(m_bTapeBuffer[m_ui32BufferIdx]);
+                        m_ui32BufferIdx++;
+                    }
 
-            // Skip passed the null terminator
-            m_ui32BufferIdx++;
+                    // Skip passed the null terminator
+                    m_ui32BufferIdx++;
+                }
+                catch (Exception)
+                {
+                    programHeader = new ProgramHeader();
```

Comment:
As discussed by email: I proposed that we should return null here in the catch block, but returning an empty object is easier for integrating with existing code so is acceptable as well  as a way to alleviate impact of bad .tap files on the behaviour of the application.